polar-core/src/debugger.rs

```diff
@@ -35,89 +48,14 @@ impl PolarVirtualMachine {
-    /// debug(), b() # Body of rule a().
-    /// b()          # Second term in the body of rule a().
-    /// ```
-    Over {
-        /// Snapshot of the current query stack sans the current query.
-        snapshot: Queries,
-    },
-    /// Step **out** of the current parent query, evaluating goals until reaching the
-    /// [`Goal::Query`](../vm/enum.Goal.html) for its next sibling.
-    ///
-    /// To illustrate this movement, let's step through the queries for the following snippet of
-    /// polar:
-    ///
-    /// ```text
-    /// a() if b() and c();
-    /// b() if debug() and d();
-    /// c();
-    /// d();
-    /// ?= a()
-    /// ```
-    ///
-    /// First we query for the head of `a()`, then the body of `a()`, and then `b()`, the first
-    /// clause in the body of `a()`. Next we query for the body of `b()` and the first clause in
-    /// the body of `b()`, a `debug()` predicate. By the time we reach that `debug()` predicate,
-    /// the query stack looks like this:
-    ///
-    /// ```text
-    /// a()          # head of a()
-    /// b(), c()     # body of a()
-    /// b()          # first clause in body of a() / head of b()
-    /// debug(), d() # body of b()
-    /// debug()      # first clause in body of b()
-    /// ```
-    ///
-    /// From our current position (evaluating the `debug()` predicate in the body of `b()`), we
-    /// would [`Step::Out`](enum.Step.html) if we wanted to continue evaluating goals until
-    /// reaching `c()` in the body of `a()`. We're stepping entirely **out** of the current parent
-    /// query, `b()`, to arrive at its next sibling, `c()`. When we arrive at the
-    /// [`Goal::Query`](../vm/enum.Goal.html) for `c()`, the query stack will look as follows:
-    ///
-    /// ```text
-    /// a()          # head of a()
-    /// b(), c()     # body of a()
-    /// c()          # second clause in body of a() / head of c()
-    /// ```
-    ///
-    /// The query stack above `c()` is identical to the previous stack snippet above `b()`, which
-    /// makes sense since they share a parent -- the query for the body of `a()`. That gives us our
-    /// test for stepping **out**:
-    ///
-    /// - Store a slice of the current query stack *without the last three queries* (current query,
-    ///   body of current rule, and head of current rule).
-    /// - Evaluate goals until the current query stack *without the current query* matches the
-    ///   stored slice.
-    Out {
-        /// Snapshot of the current query stack sans its last three queries.
-        snapshot: Queries,
-    },
+    /// Step **over** the current query. Will break on the next query where the trace stack is at the level
+    /// as the current one.
+    Over { level: usize },
+    /// Step **out** of the current query. Will break on the next query where the trace stack is at a lower
+    /// level than the current one.
+    Out { level: usize },
+    /// Step **in**. Will break on the next query.
+    InTo,
```

Comment:
nit: Can we capitalize this variant as `Into`?

Reply:
I did this because vscode was hi-lighting it differently because Into is a builtin rust thing.

Reply:
updated it to `Into`